app/src/main/java/org/hwyl/sexytopo/control/graph/GraphView.java

```diff
@@ -1182,40 +1183,57 @@ private void drawSketch(Canvas canvas, Sketch sketch, int alpha) {
             return;
         }
 
+        drawPaint.setAlpha(alpha);
+
+        int lastColour = -1;
         for (PathDetail pathDetail : sketch.getPathDetails()) {
 
             if (!couldBeOnScreen(pathDetail)) {
                 continue;
             }
 
-            drawPaint.setColor(pathDetail.getColour().intValue);
-            drawPaint.setAlpha(alpha);
+            // Avoiding constantly updating the paint colour saves approx. 10% of sketch draw time.
+            // Ideally getPathDetails() would return the paths in colour order but HashSets
+            // are unordered collections
+            if (pathDetail.getColour().intValue != lastColour) {
```

Comment:
That seems a bit crazy that this saves any time - if `setColor()` is such an expensive operation you would have thought Android would have a check to do nothing if it's already got the right colour! But I guess not...

Reply:
Yeah it shocked me but the profiler showed the difference pretty accurately